x/twap/logic_test.go

```diff
@@ -43,130 +43,278 @@ func newExpRecord(accum0, accum1 sdk.Dec) types.TwapRecord {
 	}
 }
 
+func (s *TestSuite) TestNewTwapRecord() {
+	// prepare pool before test
+	poolId := s.PrepareBalancerPoolWithCoins(defaultTwoAssetCoins...)
+
+	tests := map[string]struct {
+		poolId        uint64
+		denom0        string
+		denom1        string
+		expectedErr   error
+		expectedPanic bool
+	}{
+		"denom with non-lexicographical order": {
+			poolId,
+			denom0,
+			denom1,
+			nil,
+			false,
+		},
+		"denom with lexicographical order": {
+			poolId,
+			denom1,
+			denom0,
+			nil,
+			false,
+		},
```

Comment:
This is still backwards, non-lexicographical order is denom1, denom0. Lexicographical order is denom0, denom1.

Reply:
must have missed it! Thanks for double checking!